src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -0,0 +1,526 @@
+}
+
+std::optional<LightData> AssimpImporter::doLight(UnsignedInt id) {
+    const aiLight* l = _scene->mLights[id];
+
+    LightData::Type lightType;
+    if(l->mType == aiLightSource_DIRECTIONAL) {
+        lightType = LightData::Type::Infinite;
+    } else if(l->mType == aiLightSource_POINT) {
+        lightType = LightData::Type::Point;
+    } else if(l->mType == aiLightSource_SPOT) {
+        lightType = LightData::Type::Spot;
+    } else {
+        Error() << "Trade::AssimpImporter::light(): Undefined light type is not supported.";
+        return {};
+    }
+
+    Color3 ambientColor = Color3::from(reinterpret_cast<const float*>(&l->mColorDiffuse));
+
+    /** @todo angle inner/outer cone, linear/quadratic/constant atteniution, diffuse/specular color are not used */
+    return LightData(lightType, ambientColor, 1.0f, l);
+}
+
+UnsignedInt AssimpImporter::doMesh3DCount() const {
+    return _scene->mNumMeshes;
+}
+
+std::optional<MeshData3D> AssimpImporter::doMesh3D(const UnsignedInt id) {
+    const aiMesh* mesh = _scene->mMeshes[id];
+
+    /* Primitive */
+    MeshPrimitive primitive;
+    if(mesh->mPrimitiveTypes == aiPrimitiveType_POINT) {
+        primitive = MeshPrimitive::Points;
+    } else if(mesh->mPrimitiveTypes == aiPrimitiveType_LINE) {
+        primitive = MeshPrimitive::Lines;
+    } else if(mesh->mPrimitiveTypes == aiPrimitiveType_TRIANGLE) {
+        primitive = MeshPrimitive::Triangles;
+    } else {
+        Error() << "Trade::AssimpImporter::mesh3D(): Unsupported aiPrimitiveType.";
+        return std::nullopt;
+    }
+
+    /* Mesh data */
+    std::vector<std::vector<Vector3>> positions;
+    std::vector<std::vector<Vector2>> textureCoordinates;
+    std::vector<std::vector<Vector3>> normals;
+    std::vector<std::vector<Color4>> colors;
+
+    /* Import positions */
+    positions.emplace_back(); // FIXME(C++17) assign to pos directly
+    auto& pos = positions[0];
+    pos.reserve(mesh->mNumVertices);
+    pos.insert(pos.end(), reinterpret_cast<Vector3*>(mesh->mVertices),
+            reinterpret_cast<Vector3*>(mesh->mVertices) + mesh->mNumVertices);
+
+    normals.emplace_back(); // FIXME(C++17) assign to pos directly
+    auto& norms = normals[0];
+    norms.reserve(mesh->mNumVertices);
+    norms.insert(norms.end(), reinterpret_cast<Vector3*>(mesh->mNormals),
+            reinterpret_cast<Vector3*>(mesh->mNormals) + mesh->mNumVertices*sizeof(Vector3));
+
+    /** @todo only first uv layer (or "channel") supported) */
+    textureCoordinates.reserve(mesh->GetNumUVChannels());
+    for (size_t layer = 0; layer < mesh->GetNumUVChannels(); ++layer) {
+        textureCoordinates.emplace_back();
+        auto& texCoords = textureCoordinates[layer];
+        texCoords.reserve(mesh->mNumVertices);
+        for(size_t i = 0; i < mesh->mNumVertices; ++i) {
+            /** GCC 4.7 has a problem with .x/.y here */
+            texCoords.emplace_back(mesh->mTextureCoords[layer][i][0], mesh->mTextureCoords[layer][i][1]);
```

Comment:
MeshData doesn't support 3D texture coords, so I though, this at least gives as much information from the file as possible... but, yes there is numUVComponents in the aiMesh structure